src/track/cue.h

```diff
@@ -17,21 +17,23 @@ class Cue : public QObject {
     Q_OBJECT
 
   public:
-    enum CueSource {
-        UNKNOWN   = 0,
-        AUTOMATIC = 1,
-        MANUAL    = 2,
+    enum class Source {
+        Unknown   = 0,
+        Automatic = 1,
+        Manual    = 2,
     };
 
-    enum CueType {
-        INVALID = 0,
-        CUE     = 1, // hot cue
-        LOAD    = 2, // the cue
-        BEAT    = 3,
-        LOOP    = 4,
-        JUMP    = 5,
-        INTRO   = 6,
-        OUTRO   = 7,
+    enum class Type {
+        Invalid = 0,
+        Hotcue  = 1,
+        MainCue = 2,
```

Comment:
I think using just "Cue" or *The* Cue is too confusing if we are also using the word "cue" with a more general meaning with the name of this class.

Reply:
Probably yes, but I think DJs are used to this. Do you have you a external reference for "Main Cue"?

Reply:
No, "main cue" is a term I came up with. I am not aware of other DJ software or hardware that uses it. When I have used this language supporting users, I don't think anyone has been confused by it.

Reply:
I actually prefer the term 'LoadCue' because it clearly summarizes the purpose: The desired initial position of the track after loading. I can't think of any other use case for a 'MainCue', so why not explicitly name this use case instead of using a generic name?

Reply:
"LoadCue" only makes sense if the user has the loading preference option set to go to the cue point when tracks are loaded, which is only one option among others. Also, because there are various ways the cue point works with the various modes we have available, there are various different ways to use it, so I think only a generic name can encompass them all.

Reply:
A related discussion is here: https://github.com/mixxxdj/mixxx/pull/2273#discussion_r322899339

Reply:
OK, so lets go with MainCue